File: Microsoft.VisualStudio.ServiceModel.DomainServices.Tools/Test/Baselines/EF_Northwind_POCO.metadata.vb

Option Compare Binary
Option Infer On
Option Strict On
Option Explicit On

Imports NorthwindPOCOModel
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.ComponentModel.DataAnnotations
Imports System.Linq
Imports System.ServiceModel.DomainServices.Hosting
Imports System.ServiceModel.DomainServices.Server

Namespace NorthwindPOCOModelBuddy
    
    'The MetadataTypeAttribute identifies CategoryBuddyMetadata as the class
    ' that carries additional metadata for the CategoryBuddy class.
    <MetadataTypeAttribute(GetType(CategoryBuddy.CategoryBuddyMetadata))>  _
    Partial Public Class CategoryBuddy
        
        'This class allows you to attach custom attributes to properties
        ' of the Category class.
        '
        'For example, the following marks the Xyz property as a
        ' required property and specifies the format for valid values:
        '    <Required()>
        '    <RegularExpression("[A-Z][A-Za-z0-9]*")>
        '    <StringLength(32)>
        '    Public Property Xyz As String
        Friend NotInheritable Class CategoryBuddyMetadata
            
            'Metadata classes are not meant to be instantiated.
            Private Sub New()
                MyBase.New
            End Sub
            
            Public Property CategoryID As Integer
            
            Public Property CategoryName As String
            
            Public Property Description As String
            
            Public Property Picture() As Byte
            
            Public Property Products As List(Of Product)
        End Class
    End Class
    
    'The MetadataTypeAttribute identifies ProductBuddyMetadata as the class
    ' that carries additional metadata for the ProductBuddy class.
    <MetadataTypeAttribute(GetType(ProductBuddy.ProductBuddyMetadata))>  _
    Partial Public Class ProductBuddy
        
        'This class allows you to attach custom attributes to properties
        ' of the Product class.
        '
        'For example, the following marks the Xyz property as a
        ' required property and specifies the format for valid values:
        '    <Required()>
        '    <RegularExpression("[A-Z][A-Za-z0-9]*")>
        '    <StringLength(32)>
        '    Public Property Xyz As String
        Friend NotInheritable Class ProductBuddyMetadata
            
            'Metadata classes are not meant to be instantiated.
            Private Sub New()
                MyBase.New
            End Sub
            
            Public Property Category As Category
            
            Public Property CategoryID As Nullable(Of Integer)
            
            Public Property Discontinued As Boolean
            
            Public Property ProductID As Integer
            
            Public Property ProductName As String
            
            Public Property QuantityPerUnit As String
            
            Public Property ReorderLevel As Nullable(Of Short)
            
            Public Property SupplierID As Nullable(Of Integer)
            
            Public Property UnitPrice As Nullable(Of Decimal)
            
            Public Property UnitsInStock As Nullable(Of Short)
            
            Public Property UnitsOnOrder As Nullable(Of Short)
        End Class
    End Class
End Namespace

File: src/VisualStudio/Tools/Test/Baselines/EF_Northwind_POCO.metadata.cs

namespace NorthwindPOCOModelBuddy
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.ComponentModel.DataAnnotations;
    using System.Linq;
    using NorthwindPOCOModel;
    using OpenRiaServices.Hosting;
    using OpenRiaServices.Server;
    
    
    // The MetadataTypeAttribute identifies CategoryBuddyMetadata as the class
    // that carries additional metadata for the CategoryBuddy class.
    [MetadataTypeAttribute(typeof(CategoryBuddy.CategoryBuddyMetadata))]
    public partial class CategoryBuddy
    {
        
        // This class allows you to attach custom attributes to properties
        // of the Category class.
        //
        // For example, the following marks the Xyz property as a
        // required property and specifies the format for valid values:
        //    [Required]
        //    [RegularExpression("[A-Z][A-Za-z0-9]*")]
        //    [StringLength(32)]
        //    public string Xyz { get; set; }
        internal sealed class CategoryBuddyMetadata
        {
            
            // Metadata classes are not meant to be instantiated.
            private CategoryBuddyMetadata()
            {
            }
            
            public int CategoryID { get; set; }
            
            public string CategoryName { get; set; }
            
            public string Description { get; set; }
            
            public byte[] Picture { get; set; }
            
            public List<Product> Products { get; set; }
        }
    }
    
    // The MetadataTypeAttribute identifies ProductBuddyMetadata as the class
    // that carries additional metadata for the ProductBuddy class.
    [MetadataTypeAttribute(typeof(ProductBuddy.ProductBuddyMetadata))]
    public partial class ProductBuddy
    {
        
        // This class allows you to attach custom attributes to properties
        // of the Product class.
        //
        // For example, the following marks the Xyz property as a
        // required property and specifies the format for valid values:
        //    [Required]
        //    [RegularExpression("[A-Z][A-Za-z0-9]*")]
        //    [StringLength(32)]
        //    public string Xyz { get; set; }
        internal sealed class ProductBuddyMetadata
        {
            
            // Metadata classes are not meant to be instantiated.
            private ProductBuddyMetadata()
            {
            }
            
            public Category Category { get; set; }
            
            public Nullable<int> CategoryID { get; set; }
            
            public bool Discontinued { get; set; }
            
            public int ProductID { get; set; }
            
            public string ProductName { get; set; }
            
            public string QuantityPerUnit { get; set; }
            
            public Nullable<short> ReorderLevel { get; set; }
            
            public Nullable<int> SupplierID { get; set; }
            
            public Nullable<decimal> UnitPrice { get; set; }
            
            public Nullable<short> UnitsInStock { get; set; }
            
            public Nullable<short> UnitsOnOrder { get; set; }
        }
    }
}
